### ISSUE TITLE
Some questions regarding implementation

### ISSUE THREAD

**Issue by dshahrokhian** [harveyslash/Facial-Similarity-with-Siamese-Networks-in-Pytorch#4]
Hi, I was wondering why do you apply Dropout2D (which drops a colour channel given certain probability) in the network. As far as I know, you convert the images to single channel greyscale before feeding these to the network, which means that Dropout2D is getting rid of the single channel you have 20% of the times.

Also, what is the difference when it comes to learning improvement when using Reflective Padding?

Thanks, great tutorial and repo by the way.

Dani

**Comment by harveyslash** [comment]
Thanks for pointing it out. 
Intuitively , I assumed that individual pixels will be zeroed out , and not entire channels. 
After reading the docs, it is actually happening the way you mentioned. 

In the case of grayscale images, this effectively does nothing but slow down learning. 

If you want to submit a pull request without the dropout layers removed, feel free. 
Otherwise I will just update it myself. 

The performance was fairly similar , and I could not notice much difference. I tend to use reflection padding as it is more intuitive than just zero padding.

**Comment by dshahrokhian** [comment]
Done. Regarding reflective padding, I was mostly concerned on why is it necessary. What is the difference of applying it to just changing the input of the first fully connected layer to:

`nn.Linear(int(7.238*100*160), 500)`

Out of curiosity I found an interesting [discussion](https://twitter.com/karpathy/status/720622989289644033?lang=en) regarding the topic of Zero Padding vs Reflective Padding.

Dani

**Comment by harveyslash** [comment]
If padding is not put , the later convolution layers will gradually lose some amount of information. The final padding may not be required, but the others are (to preserve information)
